pymc3/distributions/discrete.py

```diff
@@ -689,9 +697,18 @@ def random(self, point=None, size=None, repeat=None):
         return reshape_sampled(sampled, size, self.shape)
 
     def logp(self, value):
-        return tt.switch(value > 0,
-                         tt.log(self.psi) + self.bin.logp(value),
-                         tt.log((1. - self.psi) + self.psi * tt.pow(1 - self.p, self.n)))
+        psi = self.psi
+        p = self.p
+        n = self.n
+
+        logp_val = tt.switch(value > 0,
+                 logsumexp(tt.log(psi) + self.bin.logp(value)),
+                 logsumexp(tt.log((1. - psi) + psi * tt.pow(1 - p, n))))
```

Comment:
I think this should be
`logaddexp(tt.log1p(-psi), tt.log(psi) + n * tt.log1p(-p))`
where
```python
def logaddexp(a, b):
    diff = b - a
    return tt.switch(diff > 0, b + tt.log1p(tt.exp(-diff)), a + tt.log1p(tt.exp(diff)))
```